packages/api-server/src/base/env-config.ts

```diff
@@ -41,3 +45,14 @@ function getRequired(name: string): string {
 function getOptional(name: string): string | undefined {
   return env[name];
 }
+
+export function calculateChainId(
+  creator_id: number,
+  compatible_chain_id: number
+): string {
+  const chain_id_num = compatible_chain_id * Math.pow(2, 32) + creator_id;
+  console.log(
+    `web3 chain_id: ${chain_id_num}, calculating from compatible_chain_id: ${compatible_chain_id}, creator_id: ${creator_id}`
+  );
+  return BigInt(chain_id_num).toString(10);
+}
```

Comment:
```suggestion
export function calculateChainId(
  creatorId: number,
  compatibleChainId: number
): string {
  const chainId = BigInt(compatibleChainId) << 32n + BigInt(creatorId);
  console.log(
    `web3 chain_id: ${chainId}, calculating from compatible_chain_id: ${compatibleChainId}, creator_id: ${creatorId}`
  );
  return chainId.toString(10);
}
```

And we should avoid using snake_cake in typescript if not necessary